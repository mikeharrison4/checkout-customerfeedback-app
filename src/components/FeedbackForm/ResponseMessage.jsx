import React, { useEffect } from 'react';

const ResponseMessage = ({ message, isError }) => {
  const [showPopup, setShowPopup] = React.useState(true);

  React.useEffect(() => {
    setTimeout(() => {
      setShowPopup(false);
    }, 3000);
  }, []);

  if (showPopup) {
    return (
      <div className={`absolute top-0 right-0 m-3 text-white ${isError ? 'bg-pink-500' : 'bg-green-500'}`}>
        <div className="flex justify-between p-3">
          { message }
        </div>
      </div>
    );
  }

  return null;
};

export default ResponseMessage;
