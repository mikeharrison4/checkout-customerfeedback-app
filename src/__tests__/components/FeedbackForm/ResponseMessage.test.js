import React from 'react';
import { shallow } from 'enzyme';
import ResponseMessage from '../../../components/FeedbackForm/ResponseMessage';
import { act } from 'react-dom/test-utils';

let useState;
let useEffect;

const mockUseEffect = () => {
  useEffect.mockImplementationOnce(f => f());
};

describe('ResponseMessage.jsx', () => {
  let props;
  const setState = jest.fn();

  beforeEach(() => {
    props = {
      isError: false,
      message: 'mockMessage'
    };
    jest.useFakeTimers();
    useState = jest.spyOn(React, 'useState');
    useEffect = jest.spyOn(React, 'useEffect');
    useState.mockImplementation((init) => [init, setState]);
    mockUseEffect();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should call useEffects setTimeout and turn showPopup to false', () => {
    shallow(<ResponseMessage {...props} />);
    act(() => {
      jest.runOnlyPendingTimers();
      expect(setState).toHaveBeenNthCalledWith(1, false);
    });
  });

  it('should render correctly when useState showPopup is true', () => {
    const renderedComponent = shallow(<ResponseMessage {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render correctly when useState showPopup is true and isError is true', () => {
    props.isError = true;
    const renderedComponent = shallow(<ResponseMessage {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render null when useState showPopup is false', () => {
    useState.mockImplementation(() => [false, setState]);
    const renderedComponent = shallow(<ResponseMessage {...props} />);
    expect(renderedComponent.html()).toEqual(null);
  });
});