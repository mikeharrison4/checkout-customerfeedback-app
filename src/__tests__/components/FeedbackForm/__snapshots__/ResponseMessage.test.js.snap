// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseMessage.jsx should render correctly when useState showPopup is true 1`] = `
<div
  className="absolute top-0 right-0 m-3 text-white bg-green-500"
>
  <div
    className="flex justify-between p-3"
  >
    mockMessage
  </div>
</div>
`;

exports[`ResponseMessage.jsx should render correctly when useState showPopup is true and isError is true 1`] = `
<div
  className="absolute top-0 right-0 m-3 text-white bg-pink-500"
>
  <div
    className="flex justify-between p-3"
  >
    mockMessage
  </div>
</div>
`;
